### PR TITLE
Add unit test for search iterator

### LINKED ISSUE #546: Search iterator is not being checked by unit tests
The unit test base currently does not cover the search iterator. Its addition can ease the testing of several major upcoming changes.

### PR DESCRIPTION
**Pull request by maneeshpm**
Fixes #546 

Changes contained this PR:
- Move helper code `TempZimArchive` & `TestItem` to tools for better usability.
- Add some unit tests for the search iterator.

### CODE REVIEW

**Comment by maneeshpm** [comment]
@mgautierfr why is the build failing only on some systems?

**Comment by kelson42** [comment]
@maneeshpm Thx for this PR, we have reached now 80% of code coverage :)

**Comment by kelson42** [comment]
@maneeshpm please rebase on master

**Comment by maneeshpm** [comment]
@kelson42  done.

File: test/meson.build [hunk 1]
```diff
@@ -19,6 +19,7 @@ tests = [
     'parseLongPath',
     'random',
     'tooltesting',
+    'search_iterator'
```
Comment (mgautierfr) [inline test/meson.build:22]: This test must be compiled only if `xapian_dep` is found.

**Review by mgautierfr — CHANGES_REQUESTED** [review]
It would be good to test more than just `get_title` method on the iterator.
But we can go without it for now and add it later, this is a improvement anyway.

**Comment by mgautierfr** [comment]
Please rebase-fixup the PR and we are good.

**Comment by maneeshpm** [comment]
@mgautierfr Done!

**Review by mgautierfr — APPROVED** [review]

**Pull request merged**